src/Adapter/Order/QueryHandler/GetOrderForViewingHandler.php

```diff
@@ -218,7 +218,7 @@ private function getOrderCustomer(Order $order): ?OrderCustomerForViewing
         $genderName = '';
 
         if (Validate::isLoadedObject($gender)) {
-            $genderName = $gender->name[$order->id_lang];
+            $genderName = $gender->name[$order->id_lang] ?? $gender->name[$this->contextLanguageId];
```

Comment:
Just curious, shouldn't your modif in ObjectModel prevent this?

Reply:
Oh no it only checks if `$id_lang` was provided as a parameter

Then maybe we could force the parameter in `$gender = new Gender($customer->id_gender, $order->id_lang)` However I believe when we specify the lang parameter the multilang fields are not arrays any more but converted into simple strings

So you would need to use `genderName = $gender->name` then. Maybe not the best idea 🤷‍♂️